tests/test_tocdirective.py

```diff
@@ -32,44 +31,57 @@ def test_toc_startwithlist(cli: CliRunner, temp_with_override, file_regression):
             "-W",
         ],
     )
+    # print(result.output)
     assert result.exit_code == 0
 
     path_toc_directive = path_output.joinpath("_build", "html", "index.html")
-
+    # print(path_toc_directive.read_text(encoding="utf8"))
```

Comment:
Couldn't much of this testing infra be delegated to `sphinx-external-toc` as well?

Reply:
It already is, there is loads of tests in `sphinx-external-toc` covering all this.
But I want to keep near enough all the current tests in for this PR, to keep a record that all the existing tests still pass and/or where they have changed, e.g. in the format of the toc file and the output of the `tableofcontents` directive (as I've already noted in comments here that @mmcky has made note of for potential changes to jupyterbook-latex).

In a later PR we can then certainly remove/restrucuture some of these